plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiMetadataProcessor.kt

```diff
@@ -166,6 +166,9 @@ private fun KmDeclarationContainer.removePrivateDeclarations() {
     functions.removeIf { it.visibility.isPrivate }
     properties.removeIf { it.visibility.isPrivate }
 
+    functions.sortBy(KmFunction::name)
+    properties.sortBy(KmProperty::name)
```

Comment:
```suggestion
    functions.sortWith(compareBy(KmFunction::name, { it.signature.toString() }))
    properties.sortWith(compareBy(KmProperty::name, { it.getterSignature.toString() }))
```
Sorting by name isn't enough, please also check the signatures. Also, please add a test that would check this.

Reply:
Signature is `null` in my tests used `descriptor` instead.

Reply:
> Signature is `null` in my tests used `descriptor` instead.

Probably this is related to the change in `JvmAbiOutputExtension.kt`, where it is correct.

But here we should still use the whole JVM signature, name + desc, not just the desc part. Because technically you can have declarations with the same Kotlin name (which is returned by `KmFunction.name`) and JVM signature (`JvmMethodSignature.descriptor`), but with **different** JVM names (`JvmMethodSignature.name`).

This is why my suggested change here is to use `it.signature.toString()` instead of only `it.signature?.descriptor`.

Reply:
Fixed in https://github.com/JetBrains/kotlin/pull/5233/commits/2cff02a628c05e6cfac90311d52ab733d535f311

Reply:
Thank you for the education! I assumed that signature's name is always the same as KmFunction name! From the top of my head I can't come up with a test case for that tho =( Is it about `kotlin.jvm.JvmName` ?

Reply:
Yes, `JvmName`, as well as `Deprecated(HIDDEN)` or some other mechanism to suppress the conflicting overloads error. I also can't come up with an example though...